src/plugins/drawing/components/drawing-scale-context.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, useContext } from "react";
+
+/**
+ * Represents the current SVG scaling in effect in X and Y directions due to groups.
+ */
+export interface DrawingScale {
+  scaleX: number;
+  scaleY: number;
+}
+
+/**
+ * React context for the current drawing scale. Default is no scale (1, 1).
+ */
+export const DrawingScaleContext = createContext<DrawingScale>({ scaleX: 1, scaleY: 1 });
+
+/**
+ * Hook to access the current drawing scale from context.
+ */
+export const useDrawingScale = () => useContext(DrawingScaleContext);
+
+/**
+ * Provider that multiplies the parent scale with the local scale and provides it to children.
+ * @param scale - The local scale to apply (will be multiplied with parent scale)
+ * @param children - The children to render
+ */
+export const DrawingScaleProvider:
+    React.FC<{ scale: DrawingScale; children: React.ReactNode }> = ({ scale, children }) => {
+  const parentScale = useDrawingScale();
```

Comment:
Cool - I learned something new - I didn't realize you could reach up the component tree and reuse the hook for the current provider to get the parent provider's value (which defaults to the default value of 1/1 because of the default context right?)

Reply:
Should I be embarrased to admit this?  This context method was 100% AI written.

Reply:
![image](https://github.com/user-attachments/assets/e41733b2-8502-4a23-b9a6-03cdeaccbc5e)